Sources/Widgets/Widgets3D/LineWidget/index.js

```diff
@@ -210,8 +209,13 @@ export function extend(publicAPI, model, initialValues = {}) {
     'manipulator',
     'handle1Shape',
     'handle2Shape',
+    'handle1Visibility',
+    'handle2Visibility',
+    'handle1CameraOrientation',
+    'handle2CameraOrientation',
     'positionOnLine',
```

Comment:
Can you please explain when a variable should go into `model.abc` and when it should go to `state.abc` ?

Reply:
The state.abc variables can be accessed easily within the behavior of a widget, and be shared to all the parts of the widget. The model.abc variables are set when the widget initializes and are difficult to use from the widgetProp. So for me a variable should be in the initial values if its value is needed on initialization, and a it should be placed in the state tree if it changes in the lifetime of the widget. For instance positionOnLine should be in default values since it is needed on initialization, and  it should also be in state tree since the user can interact with it to change the text position

Reply:
Thank you.

I think variables should not be both in LineWidget state AND in LineWidget model.
My understanding is LineWidget state for "display"/graphical properties and LineWidget model for "behavioral" properties.

So I believe you can clean those up.

Reply:
WDYM by "behaviorial properties"? Properties not prone to modification?

Reply:
behaviorial properties: properties that impacts the interaction with the widget.
E.g. `scaleOnRightClick`, a boolean property could carry the information to change the scale of a handle when right clicking a handle.

Reply:
so basically properties that would change somehow the state tree? Sorry but i still don't get it. A behavioral property would be a variable meant to indicate whether the state should change from one behavior to an other?

Reply:
If a property is used only by behavior.js then it is a behavioral property. 
If a property is used by a representation then it is a display property.
`handle1Visibility, handle2Visibility, handle1CameraOrientation, handle2CameraOrientation, positionOnLine` are therefore state properties and should not be added in `LineWidget/index.js`